x-pack/plugins/security_solution/public/resolver/store/selectors.ts

```diff
@@ -122,3 +132,93 @@ function composeSelectors<OuterState, InnerState, ReturnValue>(
 ): (state: OuterState) => ReturnValue {
   return (state) => secondSelector(selector(state));
 }
+
+const spatiallyIndexedEntities = createSelector(
+  (state: ResolverState) => state,
+  function (state) {
+    const indexedProcessTree = indexedProcessTreeFactory(
+      dataSelectors.graphableProcesses(state.data)
+    );
+    const processNodeViewWidth = 360;
+    const processNodeViewHeight = 120;
+    const widths = dataSelectors.widthsOfProcessSubtrees(indexedProcessTree);
+
+    const positions = dataSelectors.processPositions(indexedProcessTree, widths);
+
+    const edgeLineSegments = dataSelectors.processEdgeLineSegments(
+      indexedProcessTree,
+      widths,
+      positions
+    );
+
+    const tree: rbush<IndexedEntity> = new rbush();
+    const processesToIndex: IndexedProcessNode[] = [];
+    const edgeLineSegmentsToIndex: IndexedEdgeLineSegment[] = [];
+    for (const [processEvent, position] of positions) {
+      const transformedPosition = applyMatrix3(position, dataSelectors.isometricTransformMatrix);
+      const [nodeX, nodeY] = transformedPosition;
+      const indexedEvent: IndexedProcessNode = {
+        minX: nodeX - 0.5 * processNodeViewWidth,
+        minY: nodeY - 0.5 * processNodeViewHeight,
+        maxX: nodeX + 0.5 * processNodeViewWidth,
+        maxY: nodeY + 0.5 * processNodeViewHeight,
+        position: transformedPosition,
+        entity: processEvent,
+        type: 'processNode',
+      };
+      processesToIndex.push(indexedEvent);
+    }
+    for (const edgeLineSegment of edgeLineSegments) {
```

Comment:
It seems like most of the logic in this function is duplicated in `processNodePositionsAndEdgeLineSegments`.
Maybe instead, `processNodePositionsAndEdgeLineSegments` should return a function that takes an AABB and returns a the matching nodes and edge line segments. Then `visibleProcessNodePositionsAndEdgeLineSegments` (in this file) can get the AABB like it already does, and call `processNodePositionsAndEdgeLineSegments`.

Reasoninig:
1. Only `selectors/data` will need to know about the graph layout. This is good because the graph layout will change, and possibly end up being dynamic (in the future.)
2. No code duplication. [This code in data/selectors](https://github.com/elastic/kibana/blob/793814d0297f039ac41d31c2b8b8fa95c7d04472/x-pack/plugins/security_solution/public/resolver/store/data/selectors.ts#L546-L576) seems to be duplicated here